recipe/meta.yaml

```diff
@@ -11,12 +12,17 @@ source:
   sha256: 724f3b430dca94733aea8325cdf5a9a2043184c3bc7dff8cf6452e1848efd4e3
 
 build:
-  number: 0
+  number: {{ build_number }}
+  ignore_run_exports:
+    - cudatoolkit
+  string: cpuh{{ PKG_HASH }}_{{ build_number }}  # [cuda_compiler_version == "None"]
+  string: cuda{{ cuda_compiler_version|string }}h{{ PKG_HASH }}_{{ build_number }}  # [cuda_compiler_version != "None"]
```

Comment:
Would recommend using [a metapackage for selection]( https://github.com/conda-forge/arrow-cpp-feedstock/commit/1e57bc08f594170075e20c99db44f081e0518f63#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR21-R35 ) instead of altering the build string as the latter can be error prone